src/Compiler/SyntaxTree/LexFilter.fs

```diff
@@ -697,62 +840,75 @@ type LexFilterImpl (
     let mutable initialized = false
     let mutable offsideStack = []
     let mutable prevWasAtomicEnd = false
-    
-    let peekInitial() =
-        let initialLookaheadTokenTup = popNextTokenTup()
-        if debug then dprintf "first token: initialLookaheadTokenLexbufState = %a\n" outputPos (startPosOfTokenTup initialLookaheadTokenTup)
-        
+
+    let peekInitial () =
+        let initialLookaheadTokenTup = popNextTokenTup ()
+
+        if debug then
+            dprintf "first token: initialLookaheadTokenLexbufState = %a\n" outputPos (startPosOfTokenTup initialLookaheadTokenTup)
+
         delayToken initialLookaheadTokenTup
         initialized <- true
-        offsideStack <- (CtxtSeqBlock(FirstInSeqBlock, startPosOfTokenTup initialLookaheadTokenTup, NoAddBlockEnd)) :: offsideStack
-        initialLookaheadTokenTup 
 
-    let warn (s: TokenTup) msg = 
-        warning(IndentationProblem(msg, mkSynRange (startPosOfTokenTup s) s.LexbufState.EndPos))
+        offsideStack <-
+            (CtxtSeqBlock(FirstInSeqBlock, startPosOfTokenTup initialLookaheadTokenTup, NoAddBlockEnd))
+            :: offsideStack
+
+        initialLookaheadTokenTup
+
+    let warn (s: TokenTup) msg =
+        warning (IndentationProblem(msg, mkSynRange (startPosOfTokenTup s) s.LexbufState.EndPos))
 
     // 'query { join x in ys ... }'
-    // 'query { ... 
+    // 'query { ...
     //          join x in ys ... }'
     // 'query { for ... do
     //          join x in ys ... }'
     let detectJoinInCtxt stack =
-        let rec check s = 
-               match s with 
-               | CtxtParen(LBRACE _, _) :: _ -> true
-               | (CtxtSeqBlock _ | CtxtDo _ | CtxtFor _) :: rest -> check rest
-               | _ -> false
-        match stack with 
+        let rec check s =
+            match s with
+            | CtxtParen (LBRACE _, _) :: _ -> true
+            | (CtxtSeqBlock _
```

Comment:
These disjunctive patterns look odd

Reply:
I removed LexFilter.fs from the formatting.  It's just got a huge nubmer of disjunctive patterns. which cause problems